lint/indexOnlyLoop_checker.go

```diff
@@ -0,0 +1,81 @@
+package lint
+
+import (
+	"fmt"
+	"go/ast"
+)
+
+//! Detects when range loop is iterating over container which has pointer-type elements (that is, any *T or reference-like type as map or chan) using only key variable (no value) and container[key] occurs more than once inside loop body
+//
+// Suggests to use for key, v := range container form.
+//
+// @Before:
+// func closeFiles(files []*os.File) {
+//     for i := range files {
+//         if files[i] != nil {
+//            files[i].Close()
+//         }
+//     }
+// }
+//
+// @After:
+// func closeFilesSuggester(files []*os.File) {
+//     for _, f := range files {
+//         if f != nil {
+//             f.Close()
+//         }
+//     }
+// }
+
+func init() {
+	addChecker(&indexOnlyLoopChecker{}, attrExperimental)
+}
+
+type indexOnlyLoopChecker struct {
+	checkerBase
+}
+
+func (c *indexOnlyLoopChecker) VisitStmt(stmt ast.Stmt) {
+	if s, ok := stmt.(*ast.RangeStmt); ok && s.Key != nil {
+		sX := s.X.(*ast.Ident).Obj
```

Comment:
This will panic for things like `for i := range x.slice` or anything except `*ast.Ident`.
Should probably use `sX, ok :=` here, to avoid that.

Reply:
Also, `ast.Object` is quite deprecated thing. You may want to use `types.Object`.
(Okay for first impl, just a notice.)

Reply:
1. fixed panics on slices, added tests for that
2. don't quite understand how I can replcae *ast.Object with types Object interface here...